content/topics/coding-horrors/tdd-horrors/_index.md

```diff
@@ -0,0 +1,199 @@
+---
+title: TDD Horrors
+ready: true
+---
+
+These are all the most common TDD problems recruits tend to come across, read and make sure you didn't make any of these or any mistakes at all.
+
+- **Write tests**. Recruits in the past have done the mistake of not writing tests for TDD assignments, make sure you avoid this weird mistake. always write tests for your TDD projects.
+- Please make sure you understand .gitignore, please don't add your node_modules to git.
+- Name your files according to what is inside them
+- Avoid making tests depend on each other, either explicitly or implicitly. Dependencies among tests are a path to pain, expense, fragility, and complication.
+- Pay attention to failure messages. Make each failure message as helpful for diagnosis as you can.
+- Do not skimp on the refactoring. It is the refactoring that will keep your code (including the tests) easy to understand and change
+- Naming conventions: in general, be careful with your naming conventions. make sure your naming convention consistent, names should be more descriptive.
+- Keep good directory structure and delete all junk files.
+- Avoid messy indentation (install prettier).
+- Test for errors/exception, to do that your tests need to use:
```

Comment:
This sentence can be clearer:

sometimes code is supposed to throw an error or raise an exception. In these cases your tests should make sure that the error happens as it should.

Something like that